katdal/lazy_indexer.py

```diff
@@ -330,3 +330,39 @@ def dtype(self):
         """Type of data array after transformation, i.e. `self[:].dtype`."""
         return reduce(lambda dtype, transform: transform.dtype if transform.dtype is not None else dtype,
                       self.transforms, self._initial_dtype)
+
+
+class DaskLazyIndexer(object):
+    """Turn a dask Array into a LazyIndexer by computing it upon indexing."""
+    def __init__(self, dataset, keep=(), transforms=None):
+        self.name = getattr(dataset, 'name', '')
+        try:
+            dataset = dataset[keep]
+        except NotImplementedError:
+            # Dask does not like multiple boolean indices: go one dim at a time
+            for dim, keep_per_dim in enumerate(keep):
+                index = dataset.ndim * [slice(None)]
+                index[dim] = keep_per_dim
+                dataset = dataset[tuple(index)]
```

Comment:
I think this is reimplementing `da.take(dataset, index, axis=dim)`.

Reply:
Good point - I've never used `take` before, hence the ignorance :-)

Reply:
BTW, hopefully this workaround will soon be superfluous if dask arrays get around to to outer indexing aka oindex.

Reply:
Hmm, I might do this later as it would make dask an install requirement for katdal (it's only a test requirement and needed for the optional v4 stuff for now).